feathr_project/feathr/utils/_envvariableutil.py

```diff
@@ -3,14 +3,27 @@
 from loguru import logger
 from feathr.secrets.akv_client import AzureKeyVaultClient
 from azure.core.exceptions import ResourceNotFoundError
+from feathr.secrets.aws_secretmanager import AWSSecretManagerClient
 
 class _EnvVaraibleUtil(object):
-    def __init__(self, config_path):
+    def __init__(self, config_path: str, secret_manager_client = None):
+        """Initialize the environment variable utils client
+
+        Args:
+            config_path (str): configuration path, if users want to use YAML to load all the configs
+            secret_manager_client: the secret manager client type. currently only Azure key vault and AWS secret manager is supported.
+        """
         self.config_path = config_path
         # Set to none first to avoid invalid reference
-        self.akv_name = None
-        self.akv_name = self.get_environment_variable_with_default( 'secrets', 'azure_key_vault', 'name')
-        self.akv_client = AzureKeyVaultClient(self.akv_name) if self.akv_name else None
+        self.secret_manager_client = None
+        if secret_manager_client and self.get_environment_variable_with_default('secrets', 'azure_key_vault', 'name'):
+            self.secret_manager_client = AzureKeyVaultClient(
+                secret_namespace=self.get_environment_variable_with_default('secrets', 'azure_key_vault', 'name'),
+                secret_client=secret_manager_client)
+        elif secret_manager_client and self.get_environment_variable_with_default('secrets', 'aws_secrets_manager', 'secret_id'):
+            self.secret_manager_client = AWSSecretManagerClient(
+                secret_namespace=self.get_environment_variable_with_default('secrets', 'aws_secrets_manager', 'secret_id'),
```

Comment:
Should this introduce an else and error out in the case if the secrets manager client is not supported by feathr?

Reply:
good catch. I'll update